src/commands/list.rs

```diff
@@ -52,7 +52,7 @@ pub fn list_archive_contents(
                 Lzma => Box::new(xz2::read::XzDecoder::new(decoder)),
                 Snappy => Box::new(snap::read::FrameDecoder::new(decoder)),
                 Zstd => Box::new(zstd::stream::Decoder::new(decoder)?),
-                Tar | Zip => unreachable!(),
+                Tar | Zip => decoder,
```

Comment:
I'd like a test to make sure we don't break this in the future.

Where it asserts that:

```
.zip.zip.zip
.zip.zip
.zip.tar
.tar.tar.tar
.tar.tar
.tar.zip
```

All generate a file with the correct path name.

Reply:
I tried but I'm blocked since `ouch` does not yet allow compressing into something like `.zip.zip` :cry: 

We'd have to do that first before making this test

Reply:
@vrmiguel it's possible
```sh
$ ouch compress README.md archive.zip
$ ouch compress archive.zip archive.zip.zip --format zip
```